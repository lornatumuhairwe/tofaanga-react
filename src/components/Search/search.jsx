import React, { Component } from 'react';
import '../../styles/css/bootstrap.min.css';
import '../../styles/css/bucketlist.css';
import { FormGroup, Button, InputGroup, FormControl } from 'react-bootstrap';
import { baseUrl } from '../../constants';

export default class Search extends Component {
  constructor(props) {
    super(props);
    this.state = { search: '', result: [] };
    this.handleSearch = this.handleSearch.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(field, event) {
    const newState = {};
    newState[field] = event.target.value;
    this.setState(newState);
  }

  search() {
    this.props.getBucketlists(`${baseUrl}/bucketlists/?q=${this.state.search}`);
  }

  handleSearch(event) {
    event.preventDefault();
    this.search();
  }

  render() {
    return (<div>
      <form className="navbar-form navbar-right search" onSubmit={this.handleSearch}>
        <FormGroup>
          <InputGroup>
            <FormControl type="text" onChange={this.handleChange.bind(this, 'search')} />
            <InputGroup.Button>
              <Button type="submit">
                <i className="glyphicon glyphicon-search" />
              </Button>
            </InputGroup.Button>
          </InputGroup>
        </FormGroup>
      </form>
    </div>);
  }
}
