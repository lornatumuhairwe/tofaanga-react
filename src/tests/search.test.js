import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../components/Search/search';


describe('search snapshot', () => {
  it('matches snapshot', () => {
    const search = renderer.create(<Search />);
    expect(search.toJSON()).toMatchSnapshot();
  },

  );
});
