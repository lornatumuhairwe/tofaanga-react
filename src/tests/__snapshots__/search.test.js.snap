// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search snapshot matches snapshot 1`] = `
<div>
  <form
    className="col-md-offset-4 col-md-4"
  >
    <div
      className="form-group"
    >
      <span
        className="input-group"
      >
        <input
          className="form-control"
          id={undefined}
          onChange={[Function]}
          type="text"
        />
        <span
          className="input-group-btn"
        >
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <i
              className="glyphicon glyphicon-search"
            />
          </button>
        </span>
      </span>
    </div>
  </form>
</div>
`;
